src/v/cloud_storage_clients/s3_client.cc

```diff
@@ -556,12 +556,23 @@ s3_client::s3_client(
 
 ss::future<result<client_self_configuration_output, error_outcome>>
 s3_client::self_configure() {
+    auto result = s3_self_configuration_result{
+      .url_style = s3_url_style::virtual_host};
+    // Oracle cloud storage only supports path-style requests
+    // (https://www.oracle.com/ca-en/cloud/storage/object-storage/faq/#category-amazon),
+    // but self-configuration will misconfigure to virtual-host style due to a
+    // ListObjects request that happens to succeed. Override for this
+    // specific case.
+    auto inferred_backend = infer_backend_from_uri(_requestor._ap);
+    if (inferred_backend == model::cloud_storage_backend::oracle) {
+        result.url_style = s3_url_style::path;
+        co_return result;
+    }
+
     // Test virtual host style addressing, fall back to path if necessary.
     // If any configuration options prevent testing, addressing style will
     // default to virtual_host.
     // If both addressing methods fail, return an error.
-    auto result = s3_self_configuration_result{
-      .url_style = s3_url_style::virtual_host};
     const auto remote_read
       = config::shard_local_cfg().cloud_storage_enable_remote_read();
```

Comment:
I understand this is not part of this PR but this code seems bad for the same reasons self test checking cluster config values was bad.

We change the startup logic based on values which are meant merely to be topic defaults and are documented as such. If I don't enable remote.read or remote.write then I can't use path based bucket access? This seems wrong.

cc @andrwng

Reply:
Yeah, I agree completely. I wrote this around the same time as the self-test with the understanding that `remote.read` and `remote.write` should govern what traffic Redpanda allows, but knowing now that that is not really the case, I would be happy to rewrite this in a follow-up.